contractcourt/channel_arbitrator.go

```diff
@@ -551,6 +567,18 @@ func (c *ChannelArbitrator) supplementIncomingContestResolver(
 		)
 	}
 
+	payload, err := c.decodePayload(htlc)
+	if err != nil {
+		log.Debugf("ChannelArbitrator(%v): cannot decode payload of "+
+			"htlc %v", c.cfg.ChanPoint, htlcPoint)
+
+		// If we've locked in an htlc with an invalid payload on our
+		// commitment tx, we don't need to resolve it.
+		r.resolved = true
```

Comment:
do we need to fail back the htlc in this case? that's what we would have done if we processed the htlc entirely off chain

Reply:
but we are on-chain now, so what does fail back look like? onion parsing is only relevant for incoming htlcs, so there is also not another side of the circuit to signal

Reply:
yes of course, you're right. since this is always an incoming channel there is nothing the fail, the remote node should propagate the failure from w/e happens on chain